### ISSUE TITLE
local Demo (created via docker compose) browser compatibility

### ISSUE THREAD

**Issue by LongShanksBraveHeartKiller** [facebookresearch/sam2#342]
No matter if I use the FF nightly or any other browser (edge, chrome) I get a message similar to this

Sorry Firefox!
This version of Firefox doesn’t support the video features we’ll need to run this demo. You can either update Firefox to the latest nightly build [here](https://wiki.mozilla.org/Nightly), or try again using Chrome or Safari.

or in chrome i get this

Uh oh, this browser isn’t supported.
This browser doesn’t support the video features we’ll need to run this demo. Try again using Chrome, Safari, or Firefox Nightly.
[Back to homepage](http://localhost:7262/)

a problem with me?

**Comment by heyoeyo** [comment]
In case it helps, both chromium v128.0.6613.119 and firefox v130.0 seem to work ok for me.

That message seems to be coming from the file:
[demo/frontend/src/routes/DemoPageWrapper.tsx](https://github.com/facebookresearch/sam2/blob/52198ead0eb13ae8270bea6ca768ef175f5bf167/demo/frontend/src/routes/DemoPageWrapper.tsx)

So worst case, you could try faking support (i.e. change the [isBrowserSupported](https://github.com/facebookresearch/sam2/blob/52198ead0eb13ae8270bea6ca768ef175f5bf167/demo/frontend/src/routes/DemoPageWrapper.tsx#L30) function so it always returns true), to double check if things will work after skipping the warning.

**Comment by heyoeyo** [comment]
> To be clear, were you able to git clone the repo, fire up docker compose build, up it, and have it just work without any issue?

Yes, I haven't tried running it directly, only through docker. The only thing I did was:
```bash
docker compose up --build
```

It doesn't seem to be using the GPU so it runs painfully slow, but otherwise, there weren't any issues building/running it. For reference, my docker/compose versions are:
```bash
$ docker --version
Docker version 27.2.0, build 3ab4256

$ docker compose version
Docker Compose version v2.29.2

```

**Comment by tungtrandt13** [comment]
Some **api** properties in windows are only available with https. So everyone should edit the browser to support this section [VideoEncoder](https://developer.mozilla.org/en-US/docs/Web/API/VideoEncoder)

**Comment by nicholasbulka** [comment]
My issue is connecting to the graphql running on port 7263. I am not familiar with graphql, so maybe I am missing something easy?

**Comment by NgdHung31** [comment]
I have a problem with using my own video, I can do this by copying my video to the demo/data/gallery folder then starting the container. But if I start the container and select Upload in the frontend I get the message:
***
Uh oh, we cannot process this video
Please upload another video, and make sure that the video’s file size is less than 70Mb.
***
Although I have checked and made sure that my video's file size is less than 70Mb.
Wondering if anyone else has the same problem

**Comment by heyoeyo** [comment]
@NgdHung31 I hadn't tried it tiill now, but I get the same problem. It doesn't seem to matter what size video is used (even 200kb gives the error). Changing the 70MB limit ([here](https://github.com/facebookresearch/sam2/blob/29267c8e3965bb7744a436ab7db555718d20391a/demo/frontend/src/common/components/gallery/useUploadVideo.ts#L29)) doesn't seem to fix it either, so maybe there's something wrong with the upload process itself.

**Comment by NgdHung31** [comment]
@heyoeyo Thanks for confirming! I've actually opened an issue about this here: [https://github.com/facebookresearch/sam2/issues/361](url)
Maybe we can continue the discussion there to keep things organized?  I've included some more details about my setup in the issue. Hopefully, the SAM2 devs will be able to figure out what's going on.
